src/explore-education-statistics-admin/src/pages/publication/components/__tests__/LegacyReleasesTable.test.tsx

```diff
@@ -0,0 +1,283 @@
+      render(
+        <Router history={history}>
+          <LegacyReleasesTable publication={testPublication} />
+        </Router>,
+      );
+      userEvent.click(
+        screen.getAllByRole('button', { name: 'Edit release' })[0],
+      );
+      await waitFor(() => {
+        expect(screen.getByText('Warning'));
+      });
+      expect(
+        screen.getByText(
+          'All changes made to legacy releases appear immediately on the public website.',
+        ),
+      );
+      expect(screen.getByRole('button', { name: 'Cancel' }));
+      expect(screen.getByRole('button', { name: 'OK' }));
+    });
+
+    test('goes to the edit page when OK is clicked', async () => {
+      const history = createMemoryHistory();
+      render(
+        <Router history={history}>
+          <LegacyReleasesTable publication={testPublication} />
+        </Router>,
+      );
+      userEvent.click(
+        screen.getAllByRole('button', { name: 'Edit release' })[0],
+      );
+      await waitFor(() => {
+        expect(screen.getByText('Warning'));
+      });
+
+      userEvent.click(screen.getByRole('button', { name: 'OK' }));
+
+      await waitFor(() => {
+        expect(history.location.pathname).toBe(
+          `/publication/${testPublication.id}/legacy-releases/${testPublication.legacyReleases[0].id}/edit`,
+        );
+      });
+    });
+  });
+
+  describe('deleting', () => {
+    test('shows a warning modal when the delete release button is clicked', async () => {
+      render(<LegacyReleasesTable publication={testPublication} />);
+      userEvent.click(
+        screen.getAllByRole('button', { name: 'Delete release' })[0],
+      );
+      await waitFor(() => {
+        expect(screen.getByText('Delete legacy release'));
+      });
+      expect(
+        screen.getByText(
+          'Are you sure you want to delete this legacy release?',
+        ),
+      );
+      expect(
+        screen.getByText(
+          'All changes made to legacy releases appear immediately on the public website.',
+        ),
+      );
+      expect(screen.getByRole('button', { name: 'Cancel' }));
+      expect(screen.getByRole('button', { name: 'Confirm' }));
+    });
+
+    test('sends the delete request and updates the releases table when confirm is clicked', async () => {
+      render(<LegacyReleasesTable publication={testPublication} />);
+      userEvent.click(
+        screen.getAllByRole('button', { name: 'Delete release' })[0],
+      );
+      await waitFor(() => {
+        expect(screen.getByText('Delete legacy release'));
+      });
+      userEvent.click(screen.getByRole('button', { name: 'Confirm' }));
+
+      expect(legacyReleaseService.deleteLegacyRelease).toHaveBeenCalledWith(
+        testPublication.legacyReleases[0].id,
+      );
+
+      await waitFor(() => {
+        const table = screen.getByRole('table');
+        const rows = within(table).getAllByRole('row');
+        expect(rows).toHaveLength(3);
+      });
```

Comment:
Mentioned this before but don't use `getByRole` or `getAllByRole` within `waitFor` blocks. These selectors are slow and when used in `waitFor` can slow down tests significantly, or in the worst cases, even cause test failures due to timeouts.

The recommendation is to `getByText`, `getByTestId` or `getByLabel` as these are much faster and work well with `waitFor`.

In this case, we should probably assert that the legacy release has actually been removed from the table by checking for the deleted legacy release title. We could do some additional assertions on the table afterwards:

```ts
const table = within(screen.getByRole('table'));

await waitFor(() => {
  expect(table.queryByText('Legacy release 3')).not.toBeInTheDocument();
});
  
const rows = table.getAllByRole('row');
expect(rows).toHaveLength(3);
```

Reply:
Done